util/caller/resolver.go

```diff
@@ -45,8 +45,20 @@ type CallResolver struct {
 }
 
 // defaultPattern strips src/github.com/organization/project/module/submodule/file.go
-// to module/submodule/file.go.
-var defaultPattern = "src/" + strings.Repeat(strings.Join([]string{"[^", "]+", ""}, string(os.PathSeparator)), 3) + "(.*)"
+// down to module/submodule/file.go.
+var defaultPattern = func() string {
+	_, file, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("unable to look up location")
+	}
+	const sep = string(os.PathSeparator)
+	path := filepath.Dir(file)
+	// Strip to $GOPATH/src/.
+	for i := 0; i < 5; i++ {
+		path = filepath.Dir(filepath.Clean(path))
+	}
+	return path + sep + strings.Repeat(strings.Join([]string{"[^", "]+", ""}, sep), 3) + "(.*)"
```

Comment:
no, `Dir` always strips the last slash (unless the result is already `/`, which it can't in our case). I was thinking about using `filepath.Join` but it seemed weird since this is a regular expression with characters untypical for a path (plus it looks weird to use it all the time, and inconsistent part of it):

`filepath.Join(path, "[^", "]+", "", .....)`